src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/RenamerTests.cs

```diff
@@ -5,27 +5,54 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Text;
 using Xunit;
+using Moq;
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS
 {
     [ProjectSystemTrait]
     public class RenamerTests
     {
-        bool ConfirmForRenameWasCalled { get; set; }
+        [Theory]
+        [InlineData("class Foo{}", "Foo.cs", "Bar.cs")]
+        [InlineData("interface Foo { void m1(); void m2();}", "Foo.cs", "Bar.cs")]
+        [InlineData("delegate int Foo(string s);", "Foo.cs", "Bar.cs")]
+        [InlineData("partial class Foo {} partial class Foo {}", "Foo.cs", "Bar.cs")]
+        [InlineData("struct Foo { decimal price; string title; string author;}", "Foo.cs", "Bar.cs")]
+        [InlineData("enum Foo { None, enum1, enum2, enum3, enum4 };", "Foo.cs", "Bar.cs")]
+        [InlineData("namespace n1 {class Foo{}} namespace n2 {class Foo{}}", "Foo.cs", "Bar.cs")]
+        public async Task Rename_Symbol_Should_TriggerUserConfirmationAsync(string soureCode, string oldFilePath, string newFilePath)
+        {
+            var userNotificationServices = IUserNotificationServicesFactory.Create();
+            var roslynServices = IRoslynServicesFactory.Create();
+
+            await RenameAsync(soureCode, oldFilePath, newFilePath, userNotificationServices, roslynServices);
+
+            Mock.Get(userNotificationServices).Verify(h => h.Confirm(It.IsAny<string>()), Times.Once);
+            Mock.Get(roslynServices).Verify(h => h.RenameSymbolAsync(It.IsAny<Solution>(), It.IsAny<ISymbol>(), It.IsAny<string>()), Times.Never);
+        }
 
         [Theory]
         [InlineData("class Foo{}", "Foo.cs", "Bar.cs")]
         [InlineData("interface Foo { void m1(); void m2();}", "Foo.cs", "Bar.cs")]
         [InlineData("delegate int Foo(string s);", "Foo.cs", "Bar.cs")]
         [InlineData("partial class Foo {} partial class Foo {}", "Foo.cs", "Bar.cs")]
```

Comment:
As I am looking into the tests, I see that there is no assertion for the end result. Am I missing to see it? There is no assertion to check the text change. For the most part it might be testing Roslyn Rename but it helps us to make sure we renamed all the required symbols and also not rename the ones we shouldn't.